code/modules/atmospherics/machinery/components/binary_devices/pump.dm

```diff
@@ -21,51 +21,31 @@ Thus, the two variables affect pump operation are set in New():
 
 	can_unwrench = 1
 
-	var/on = 0
-	var/target_pressure = ONE_ATMOSPHERE
+	target_pressure = ONE_ATMOSPHERE
 
 	var/id = null
 
+// So we can CtrlClick without triggering the anchored message.
+/obj/machinery/atmospherics/binary/pump/can_be_pulled(user, grab_state, force, show_message)
+	return FALSE
+
 /obj/machinery/atmospherics/binary/pump/CtrlClick(mob/living/user)
-	if(!istype(user) || user.incapacitated())
-		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
-		return
-	if(!in_range(src, user) && !issilicon(usr))
-		return
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	toggle()
+	if(can_use_shortcut(user))
+		toggle(user)
 	return ..()
 
-/obj/machinery/atmospherics/binary/pump/AICtrlClick()
-	toggle()
+/obj/machinery/atmospherics/binary/pump/AICtrlClick(mob/living/silicon/user)
+	toggle(user)
 	return ..()
 
 /obj/machinery/atmospherics/binary/pump/AltClick(mob/living/user)
-	if(!istype(user) || user.incapacitated())
-		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
-		return
-	if(!in_range(src, user) && !issilicon(usr))
-		return
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	set_max()
-	return
+	if(can_use_shortcut(user))
+		set_max(user)
 
 /obj/machinery/atmospherics/binary/pump/AIAltClick()
 	set_max()
 	return ..()
```

Comment:
Because of this line in `ai.dm`, if the AI alt clicks on a pump `set_max()` will be called twice. Once with a `user` argument and once without:
https://github.com/ParadiseSS13/Paradise/blob/fe6e445760ade9158e5fcf84346b11d1cfb15d42/code/_onclick/ai.dm#L159-L160
I'd recommend just changing that to just `return` like all of the other procs there, but since it was added in 21a9d929304cc8c742369447c0c443028e879c09 7 years ago it might be tied into some other interactions, so probably best to leave that for another PR unless you can make sure there won't be any side effects.

TLDR: You can get rid of the `return ..()` and add a user argument here and on the others to make it work properly, but since it's only setting it to maximum it doesn't actually do anything bad.
```suggestion
/obj/machinery/atmospherics/binary/pump/AIAltClick(mob/user)
	set_max(user)
```

Reply:
Yeah good catche on this stuff.